h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostScoreTask.java

```diff
@@ -203,6 +204,12 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
         } catch (XGBoostError xgBoostError) {
             throw new IllegalStateException("Failed to score with XGBoost.", xgBoostError);
         } finally {
+            if (booster != null) {
+                booster.dispose();
+            }
+            if (data != null) {
+                data.dispose();
```

Comment:
Tried to find something about it in the docs but couldn't - think it would be a good idea just in case to wrap the `booster.dispose()` in `try{}finally{}` and do `data.dispose()` in the finally block? Not sure what errors it can throw, so might be an overkill.

Reply:
This is a good point. I thought about it but in my opinion, it makes sense not to suppress any error that occurs in the dispose call. If we fail to deallocate the memory and continue model building we might crash the whole JVM process. Perhaps a better solution would be to wrap it in a more exploratory exception and suggest to stop using this JVM for further training to avoid instability? WDYT?

Reply:
Oh, see what you mean! Thx